Bowling/BowlingGame.cs

```diff
@@ -0,0 +1,150 @@
+            }
+        }
+
+        public void Roll(int pins)
+        {
+            if (framesScoreBoard[CurrentFrameIndex].CanInsert())
+            {
+                framesScoreBoard[CurrentFrameIndex].Insert(pins);
+            }
+            else
+            {
+                if (CurrentFrameIndex == NumberOfFrames - 1)
+                {
+                    RollForLastBonus(pins);
+                }
+                else
+                {
+                    RollForNextFrame(pins);
+                }
+            }
+        }
+
+        private void RollForLastBonus(int pins)
+        {
+            if (CanRollForLastBonus() == true)
+            {
+                bonus = pins;
+            }
+            else
+            {
+                throw new UnfairBonusRollException(Constants.unfairBonusMessage);
+            }
+        }
+
+        private void RollForNextFrame(int pins)
+        {
+            CurrentFrameIndex += 1;
+            framesScoreBoard[CurrentFrameIndex].Insert(pins);
+        }
+
+        public void DisplayScore()
+        {
+            var numFramesExceptLast = NumberOfFrames - 1;
+            for (var frameIndex = 0; frameIndex < numFramesExceptLast; frameIndex++)
+            {
+                var frameShots = framesScoreBoard[frameIndex];
+                Console.WriteLine(String.Format(Constants.scoreDisplayFormat, frameShots.FirstShot, frameShots.SecondShot, CalculateFrameScore(frameIndex)));
+            }
+
+            var lastFrameShots = framesScoreBoard[numFramesExceptLast];
+            Console.WriteLine(String.Format(Constants.scoreDisplayFormat, lastFrameShots.FirstShot, lastFrameShots.SecondShot, CalculateLastFrameScore(numFramesExceptLast)));
+            if(bonus != -1)
+                Console.WriteLine("Last bonus: " + bonus);
+        }
+
+        private int CalculateFrameScore(int frameIndex)
+        {
+            var frameShots = framesScoreBoard[frameIndex];
+            var nextFrameShots = framesScoreBoard[frameIndex + 1];
+            if (frameShots.IsStrike())
+            {
+                return frameShots.FirstShot + frameShots.SecondShot + nextFrameShots.FirstShot;
+            }
+            else if (frameShots.IsSpare())
+            {
+                return frameShots.FirstShot + frameShots.SecondShot + nextFrameShots.FirstShot;
+            }
+            return frameShots.FirstShot + frameShots.SecondShot;
+        }
+
+        private int CalculateLastFrameScore(int frameIndex)
+        {
+            var frameShots = framesScoreBoard[frameIndex];
+            if (bonus != -1)
+            {
+                return frameShots.FirstShot + frameShots.SecondShot + bonus;
+            }
+            return frameShots.FirstShot + frameShots.SecondShot;
+        }
+
+        private bool CanRollForLastBonus()
+        {
+            var lastFrame = framesScoreBoard[CurrentFrameIndex];
+            if (bonus == -1 && (lastFrame.IsSpare() || lastFrame.IsStrike()))
```

Comment:
Here you can return directly `return bonus == -1 && (lastFrame.IsSpare() || lastFrame.IsStrike();`